main/zoe/guide/contracts/multipoolAutoswap.md

```diff
@@ -0,0 +1,177 @@
+# Multipool Autoswap
+
+<Zoe-Version/>
+
+##### [View the code on Github](https://github.com/Agoric/agoric-sdk/blob/2a8b0fc2ece7344604bcc23b295367cd871f6995/packages/zoe/src/contracts/multipoolAutoswap/multipoolAutoswap.js) (Last updated: 2020-09-14)
+##### [View all contracts on Github](https://github.com/Agoric/agoric-sdk/tree/master/packages/zoe/src/contracts)
+
+
+Multipool Autoswap is a rewrite of [UniSwap](https://uniswap.org/) that supports
+multiple liquidity pools and direct exchanges across pools.
+
+Each liquidity pool maintains a price for exchanges between the central token and a
+secondary token. Secondary tokens can be exchanged with each other, but only through
+the central token. For example, if X and Y are two token types and C is the central
+token, a swap giving X and wanting Y would first use the pool (X, C) then the pool
+(Y, C). There are no direct liquidity pools between two secondary tokens.
+
+There should only need to be one instance of this contract, so liquidity can be
+shared as much as possible. Each secondary currency has a separate pool of liquidity.
+
+When the contract is instantiated, the central token is specified in the
+terms. Separate invitations are available by calling methods on the publicFacet for
+adding and removing liquidity and for making trades. Other publicFacet operations
+support querying prices and the sizes of pools. New pools can be created with
+`addPool()`.
+
+When making trades or requesting prices, the caller must specify that either the
+input price (swapIn, getInputPrice) or the output price (swapOut, getOutPutPrice) is
+fixed. For swaps, the required keywords are `In` for the trader's `give` amount, and
+`Out` for the trader's `want` amount.  `getInputPrice()` and `getOutputPrice()` each
+take an amount for the currency that is being specified, and a brand to indicate the
+currency of the desired amount.
+
+When adding and removing liquidity, the keywords are `Central`, `Secondary`, and
+`Liquidity`. Adding liquidity uses `Central` and `Secondary` in the `give` section
+and `Liquidity` in the `want` section. Removing liquidity uses these same
+keywords, reversed: `Liquidity` in the `give` section, and `Central` and
+`Secondary` in the want section. If the proposal specifies amounts directly taken
+from a recent quote, and any trading has intervened, the trade is unlikely to be
+accepted. You can either specify limits on how far the price may have moved, or
+specify limits of zero and trust the contract to trade fairly.
```

Comment:
You can specify either limits on how far the price may have moved, or limits of zero...